llvm/lib/Target/AIE/AIEBaseInstrInfo.cpp

```diff
@@ -707,6 +707,11 @@ int AIEBaseInstrInfo::getMaxLastMemoryCycle() const {
   return std::numeric_limits<unsigned>().min();
 }
 
+std::optional<std::vector<int>>
+AIEBaseInstrInfo::getMemoryCycles(unsigned SchedClass) const {
+  return std::nullopt;
```

Comment:
nit: As this is a default basic implementation for a base class method, it does not hurt to be directly implemented in the .h file.